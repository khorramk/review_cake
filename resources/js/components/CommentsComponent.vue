<template>
    <div>
        <div class="col comments-card_single-comments__wrapper">
            <div class="card-body comment-card__single-card" >
               {{this.$props.comment.comments}}
                <a class="review-card__single-review_edit-comment right-side"  :href="`/comments/${$props.comment.id}/edit`">ðŸ–‰</a>
                <form  class="review-card__single-review_remove-comment right-side" @submit="submit($props.comment.id)">
                    <input :disabled="isDisable" type="submit" class="button-remove p-r" name="btnDelete" value="â¨¯">
                </form>
            </div>
        </div>
    </div>               
</template>

<script>
import Axios from 'axios';
    export default {
        data() {
            return {
              reviewIds: '',
              isDisable: false
            };
        },
        methods: {
            submit(id){  
                this.isDisable = true;
                Axios.delete(`/api/comments/${id}`)
                     .then(()=> window.location.href = '/api/reviews');
            },
        },
        props: ['comment']
    };
</script>
