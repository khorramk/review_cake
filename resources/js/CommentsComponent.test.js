//test.js

import Vue from 'vue';
import axios from 'axios';
import VueRouter from 'vue-router';
import CommentsComponent from './components/CommentsComponent';

const vm = new Vue(CommentsComponent);


it('renders the component', () => {
   
    vm.$nextTick().then(()=> {
 
     expect(vm).toMatchSnapshot();
    }).catch().finally()
  })