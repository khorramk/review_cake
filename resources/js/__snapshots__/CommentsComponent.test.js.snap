// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
Vue {
  "$attrs": Object {},
  "$children": Array [],
  "$createElement": [Function],
  "$listeners": Object {},
  "$options": Object {
    "_base": [Function],
    "_propKeys": Array [
      "comment",
      "reviewId",
    ],
    "components": Object {},
    "directives": Object {},
    "filters": Object {},
    "methods": Object {
      "deleteComments": [Function],
    },
    "props": Object {
      "comment": Object {
        "type": [Function],
      },
      "reviewId": Object {
        "type": [Function],
      },
    },
    "render": [Function],
    "staticRenderFns": Array [],
  },
  "$parent": undefined,
  "$refs": Object {},
  "$root": [Circular],
  "$scopedSlots": Object {},
  "$slots": Object {},
  "$vnode": undefined,
  "_c": [Function],
  "_data": Object {},
  "_directInactive": false,
  "_events": Object {},
  "_hasHookEvent": false,
  "_inactive": null,
  "_isBeingDestroyed": false,
  "_isDestroyed": false,
  "_isMounted": false,
  "_isVue": true,
  "_props": Object {
    "comment": undefined,
    "reviewId": undefined,
  },
  "_renderProxy": Vue {
    "$attrs": Object {},
    "$children": Array [],
    "$createElement": [Function],
    "$listeners": Object {},
    "$options": Object {
      "_base": [Function],
      "_propKeys": Array [
        "comment",
        "reviewId",
      ],
      "components": Object {},
      "directives": Object {},
      "filters": Object {},
      "methods": Object {
        "deleteComments": [Function],
      },
      "props": Object {
        "comment": Object {
          "type": [Function],
        },
        "reviewId": Object {
          "type": [Function],
        },
      },
      "render": [Function],
      "staticRenderFns": Array [],
    },
    "$parent": undefined,
    "$refs": Object {},
    "$root": [Circular],
    "$scopedSlots": Object {},
    "$slots": Object {},
    "$vnode": undefined,
    "_c": [Function],
    "_data": Object {},
    "_directInactive": false,
    "_events": Object {},
    "_hasHookEvent": false,
    "_inactive": null,
    "_isBeingDestroyed": false,
    "_isDestroyed": false,
    "_isMounted": false,
    "_isVue": true,
    "_props": Object {
      "comment": undefined,
      "reviewId": undefined,
    },
    "_renderProxy": [Circular],
    "_self": [Circular],
    "_staticTrees": null,
    "_uid": 0,
    "_vnode": null,
    "_watcher": null,
    "_watchers": Array [],
    "comment": undefined,
    "deleteComments": [Function],
    "reviewId": undefined,
  },
  "_self": [Circular],
  "_staticTrees": null,
  "_uid": 0,
  "_vnode": null,
  "_watcher": null,
  "_watchers": Array [],
  "comment": undefined,
  "deleteComments": [Function],
  "reviewId": undefined,
}
`;
